packages/netlify-cms-backend-github/src/API.ts

```diff
@@ -864,13 +866,37 @@ export default class API {
     const resp: { statuses: GitHubCommitStatus[] } = await this.request(
       `${this.originRepoURL}/commits/${sha}/status`,
     );
-    return resp.statuses.map(s => ({
-      context: s.context,
-      // eslint-disable-next-line @typescript-eslint/camelcase
-      target_url: s.target_url,
-      state:
-        s.state === GitHubCommitStatusState.Success ? PreviewState.Success : PreviewState.Other,
-    }));
+    const deploys: GitHubDeployments = await this.request(
+      `${this.originRepoURL}/deployments?task=deploy&ref=${sha}`,
+    );
+    const depStatuses = await Promise.all(
+      deploys.map(async dep => {
+        const stats: GitHubDeployStatuses = await this.request(
+          `${this.originRepoURL}/deployments/${dep.id}/statuses`,
+        );
+
+        return stats
+          .map(stat => ({
+            context: stat.description,
+            // eslint-disable-next-line @typescript-eslint/camelcase
+            target_url: stat.target_url,
+            state:
+              stat.state === GitHubCommitStatusState.Success
+                ? PreviewState.Success
+                : PreviewState.Other,
+          }))
+          .reduce((acc, stat) => (stat.state === PreviewState.Success ? stat : acc));
```

Comment:
```suggestion
        const statuses = stats.map(stat => ({
          context: stat.description,
          // eslint-disable-next-line @typescript-eslint/camelcase
          target_url: stat.target_url,
          state:
            stat.state === GitHubCommitStatusState.Success
              ? PreviewState.Success
              : PreviewState.Other,
        }));

        return statuses.reverse().find(s => s.state === PreviewState.Success) || statuses[0];
```

Would this be easier to read?